examples/opencv_rc.py

```diff
@@ -0,0 +1,541 @@
+
+        # Keep track of Cozmo's linear and angular velocities
+        self._velocity = {'linear': 0, 'angular': 0}
+
+        # Keep track of the head light state
+        self._head_light = False
+
+        # Keep track of the head tilt angle
+        self._head_tilt = pc.MIN_HEAD_ANGLE.radians
+
+        # Keep track of the lift's height
+        self._lift_height = pc.MIN_LIFT_HEIGHT.mm
+
+        # Initialize Cozmo's client
+        self._cozmo_clt = pc.Client()
+
+        # Initialize a listener to monitor keyboard events
+        self._kbd_listener = kbd.Listener(on_press=self._on_keypress,
+                                          on_release=self._on_keyrelease)
+
+        # Other miscellaneous parameters to keep track of
+        self._win_name = win_name
+        self._sharp_amount = sharp_amount
+        self._sharp_gamma = sharp_gamma
+
+    def init(self):
+        """
+        Create the window OpenCV will use to display the video feed and the
+        different trackbars, as well as set and start Cozmo's client, and the
+        keyboard listener.
+        :return: None
+        """
+
+        # Create a window for the control panel
+        cv.namedWindow(self._win_name)
+
+        # Create the different trackbars controlling the robot's velocities,
+        # head tilt, and lift height
+        cv.createTrackbar('Linear Velocity', self._win_name, 0, 100,
+                          self._on_linear_velocity_change)
+        cv.createTrackbar('Angular Velocity', self._win_name, 0, 100,
+                          self._on_angular_velocity_change)
+        cv.createTrackbar('Head tilt', self._win_name, 0, 100,
+                          self._on_head_tilt_change)
+        cv.createTrackbar('Lift height', self._win_name, 0, 100,
+                          self._on_lift_height_change)
+
+        # Start Cozmo's client and connect to the robot
+        self._cozmo_clt.start()
+        self._cozmo_clt.connect()
+        self._cozmo_clt.wait_for_robot()
+
+        # Set Cozmo in its initial state
+        # Look down
+        self.head_tilt = pc.MIN_HEAD_ANGLE.radians
+
+        # Set the lift in its minimum position
+        self.lift_height = pc.MIN_LIFT_HEIGHT.mm
+
+        # Make sure the light is off by default
+        self.head_light = False
+
+        # Enable the camera
+        self._cozmo_clt.enable_camera(enable=True, color=self._color)
+
+        # Handle new incoming images
+        self._cozmo_clt.add_handler(pc.event.EvtNewRawCameraImage,
+                                    self._on_new_image)
+
+        # Handle cliff and pick-up detection
+        self._cozmo_clt.add_handler(pc.event.EvtCliffDetectedChange,
+                                    self._stop_all)
+        self._cozmo_clt.add_handler(pc.event.EvtRobotPickedUpChange,
+                                    self._stop_all)
+
+        # Start the keyboard event listener
+        self._kbd_listener.start()
+        self._kbd_listener.wait()
```

Comment:
Won't this wait statement stop the main thread here?

Reply:
This is a bit confusing indeed, but according to [the documentation](https://pynput.readthedocs.io/en/latest/keyboard.html#pynput.keyboard.Listener.wait) this will only stop the main thread until the listener is ready to do its job. In essence, it has the same purpose as the `self._cozmo_clt.wait_for_robot()` function above.

Reply:
Sounds good then! I wasn't sure so I thought I'd like to ask ;)